conjure-core/src/main/java/com/palantir/conjure/defs/EndpointErrorResolver.java

```diff
@@ -0,0 +1,158 @@
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.defs;
+
+import com.google.common.base.Preconditions;
+import com.palantir.conjure.defs.EndpointErrorResolver.ErrorResolutionResult;
+import com.palantir.conjure.parser.AnnotatedConjureSourceFile;
+import com.palantir.conjure.parser.types.ConjureType;
+import com.palantir.conjure.parser.types.ConjureTypeVisitor;
+import com.palantir.conjure.parser.types.builtin.AnyType;
+import com.palantir.conjure.parser.types.builtin.BinaryType;
+import com.palantir.conjure.parser.types.builtin.DateTimeType;
+import com.palantir.conjure.parser.types.collect.ListType;
+import com.palantir.conjure.parser.types.collect.MapType;
+import com.palantir.conjure.parser.types.collect.OptionalType;
+import com.palantir.conjure.parser.types.collect.SetType;
+import com.palantir.conjure.parser.types.complex.ErrorTypeDefinition;
+import com.palantir.conjure.parser.types.names.ConjurePackage;
+import com.palantir.conjure.parser.types.names.Namespace;
+import com.palantir.conjure.parser.types.names.TypeName;
+import com.palantir.conjure.parser.types.primitive.PrimitiveType;
+import com.palantir.conjure.parser.types.reference.ForeignReferenceType;
+import com.palantir.conjure.parser.types.reference.LocalReferenceType;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Endpoint error definitions are either references to errors defined in the same file as the endpoint definition, or in
+ * imported Conjure files. This class creates a {@link com.palantir.conjure.spec.TypeName} object from a
+ * {@link com.palantir.conjure.parser.types.reference.ReferenceType} to a Conjure defined error definition.
+ */
+final class EndpointErrorResolver implements ConjureTypeVisitor<ErrorResolutionResult> {
+    private static final String UNSUPPORTED_TYPE_MESSAGE =
+            "Unsupported endpoint error type. Endpoint errors must be references to a Conjure-defined error type";
+
+    private final Map<TypeName, ErrorTypeDefinition> parsedErrors;
+    private final Optional<ConjurePackage> defaultConjurePackage;
+    private final Map<Namespace, String> importProviders;
+    private final Map<String, AnnotatedConjureSourceFile> externalTypes;
+
+    EndpointErrorResolver(
+            Map<TypeName, ErrorTypeDefinition> parsedErrors,
+            Optional<ConjurePackage> defaultConjurePackage,
+            Map<Namespace, String> importProviders,
+            Map<String, AnnotatedConjureSourceFile> externalTypes) {
+        this.parsedErrors = parsedErrors;
+        this.defaultConjurePackage = defaultConjurePackage;
+        this.importProviders = importProviders;
+        this.externalTypes = externalTypes;
+    }
+
+    record ErrorResolutionResult(
+            String errorName, String package_, com.palantir.conjure.spec.ErrorNamespace namespace) {}
```

Comment:
Would it make sense to use the new ErrorTypeName type here?

Reply:
Yep. Updated.